cmd/clusterctl/client/repository/template_client_test.go

```diff
@@ -158,36 +161,27 @@ func Test_templates_Get(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			f := newTemplateClient(tt.fields.provider, tt.fields.version, tt.fields.repository, tt.fields.configVariablesClient)
 			got, err := f.Get(tt.args.flavor, tt.args.targetNamespace, tt.args.listVariablesOnly)
-			if (err != nil) != tt.wantErr {
-				t.Fatalf("error = %v, wantErr %v", err, tt.wantErr)
-			}
 			if tt.wantErr {
+				g.Expect(err).To(HaveOccurred())
 				return
 			}
 
-			if !reflect.DeepEqual(got.Variables(), tt.want.variables) {
-				t.Errorf("got.Variables() = %v, want = %v ", got.Variables(), tt.want.variables)
-			}
+			g.Expect(err).NotTo(HaveOccurred())
```

Comment:
nit we should be consistent in having assertion on the operation err/on wantErr like in Test_componentsClient_Get